### PR TITLE
Add date picker to Progress tab period reports

### PR DESCRIPTION
**Pull request by claudiovallejo**
**Story card:** [sc-7453](https://app.shortcut.com/simpledotorg/story/7453/replace-period-report-carousels-with-date-pickers)

## Because

We need to replace the side-scrolling carousels with dropdowns. Unfortunately, side-scrolling isn't supported on the Simple App because horizontal swipes trigger swapping between tabs.

## This addresses
Changes period report header from this...
image

To this...
image

### Changes
* Removes pill date carousels
* Adds a styled dropdown to pick a date for the period reports

## Test instructions

1. Open your local and set the `new_progress_tab` Flipper flag
2. Go to any facility report page and click "Progress report"
3. Click the "Daily", "Monthly", and "Yearly" reports and test the new dropdown at the top of the page

### CODE REVIEW

File: app/views/api/v3/analytics/user_analytics/_period_report.html.erb [hunk 1]
```diff
@@ -18,13 +18,11 @@
...
-      <% periods.each do |pill| %>
-        <%= render partial: "api/v3/analytics/user_analytics/pill",
-                  locals: { title: pill,
-                            is_active: pill == periods.first ? true : false } %>
-      <% end %>
+      <select class="w-100 mt-16px p-12px fw-medium fs-18px c-blue bgc-blue-light b-none br-4px bs-secondary-button o-none a-none chevron-down-icon">
+        <% periods.each do |period| %>
+          <option value="<%= period %>"><%= period %></option>
+        <% end %>
+      </select>
```
Comment (harimohanraj89) [inline app/views/api/v3/analytics/user_analytics/_period_report.html.erb:25]: Can we use Rails's `select_tag` ([doc](https://apidock.com/rails/ActionView/Helpers/FormTagHelper/select_tag)) for this? Check out the last example in the doc using `options_for_select` that would be applicable for us here.

**Review by harimohanraj89 — CHANGES_REQUESTED** [review]
Let's Rails-ify this up a little bit.

File: app/views/api/v3/analytics/user_analytics/_period_report.html.erb (hunk 1 above)
Comment (claudiovallejo) [reply app/views/api/v3/analytics/user_analytics/_period_report.html.erb]: Done, thanks for the feedback @harimohanraj89 !

File: app/views/api/v3/analytics/user_analytics/_period_report.html.erb [hunk 2]
```diff
@@ -18,11 +18,7 @@
           <%= subtitle %>
         </p>
       <% end %>
-      <select class="w-100 mt-16px p-12px fw-medium fs-18px c-blue bgc-blue-light b-none br-4px bs-secondary-button o-none a-none chevron-down-icon">
-        <% periods.each do |period| %>
-          <option value="<%= period %>"><%= period %></option>
-        <% end %>
-      </select>
+      <%= select_tag "periods", options_for_select(periods, periods.first()), class: "w-100 mt-16px p-12px fw-medium fs-18px c-blue bgc-blue-light b-none br-4px bs-secondary-button o-none a-none chevron-down-icon" %>
```
Comment (harimohanraj89) [inline app/views/api/v3/analytics/user_analytics/_period_report.html.erb:21]: We can remove the ().

**Review by harimohanraj89 — APPROVED** [review]

**Pull request merged**